### PR TITLE
TA-1274: Test implementation

### PR DESCRIPTION
**Pull request by EvgeniiNikonets**

### CODE REVIEW

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+        {
+          "Name": "Delete default ACL Profile with Name: 'acl-default-1'",
+          "Skip": false,
+          "GenerateID": false,
+          "API": {
+            "Base": "acl-profile",
+            "Path": "__acldefault__/"
+          },
+          "Method": "DELETE",
+          "Expected": {
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:290]: Expected result in this test should be error - 400

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 2]
```diff
@@ -0,0 +1,662 @@
...
+            ]
+          },
+          "Expected": {
+            "Code": "400",
+            "Type": "json"
+          }
+        },
+        {
+          "Name": "Create ACL Profile with Leading/Trailing Space Characters in Name",
+          "Skip": false,
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:202]: Please, divide this check to 2 steps:
1. For leading space
2. For trailing space

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 3]
```diff
@@ -0,0 +1,662 @@
...
+          "Name": "Delete ACL Profile with Name: 'space_name_with_spaces'",
+          "Skip": false,
+          "GenerateID": false,
+          "API": {
+            "Base": "acl-profile",
+            "Path": "space_name_with_spaces/"
+          },
+          "Method": "DELETE",
+          "Expected": {
+            "Code": "200",
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:345]: Expected result in this test should be error - 400

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+                "Value": "\"acl_profile_special_char_tag\""
+              },
+              {
+                "Key": "id",
+                "Value": "\"acl_profile_special_char_tag\""
+              },
+              {
+                "Key": "tags",
+                "Value": "[\"tag_with_special_char\"]"
+              }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:555]: {
                "Key": "tags",
                "Value": "[\"!@#$%^&*()<>'_-.\"]"
              }

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 5]
```diff
@@ -0,0 +1,662 @@
...
+            ]
+          },
+          "Expected": {
+            "Code": "400",
+            "Type": "json"
+          }
+        },
+        {
+          "Name": "Create ACL Profile with Tag Containing Leading/Trailing Space Characters",
+          "Skip": false,
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:629]: Please, divide this check to 2 steps:

For leading space
For trailing space

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 6]
```diff
@@ -0,0 +1,662 @@
...
+                "Key": "tags",
+                "Value": "[\"  tag_with_spaces  \"]"
+              }
+            ]
+          },
+          "Expected": {
+            "Code": "200",
+            "Type": "json"
+          }
+        }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:658]: Add Step to check that tag was saved without leading/trailing space - GET acl-profile with name 'acl_profile_tag_with_spaces' and verify in Expected response that tag was trimmed 
SingleResult:[
{
                "Key": "tags",
                "Value": "[\"tag_with_spaces\"]"
              }
]

File: example_suites/add_edit_duplicate_delete_acl_policy.json (hunk 3 above)
Comment (EvgeniiNikonets) [reply example_suites/add_edit_duplicate_delete_acl_policy.json:338]: not real , that Step 6 - delete not attached policy:

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 7]
```diff
@@ -0,0 +1,655 @@
+{
+  "Name": "ACL - Add/Edit/Duplicate/Delete ACL policy TA-1274",
+  "Skip": false,
+  "Tests": [
+    {
+      "Name": "ACL - Add/Edit/Duplicate/Delete ACL policy",
+      "Skip": false,
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:7]: It looks like you did all the tests in 1 big Test. Please, create separate Test for each step in TP.
For example:
The first Test is "'Name' field validations":
Steps:
1. Create ACL Profile with Valid Name
2. Create ACL Profile with Empty Name
3. ...
The second Test is "Edit ACL policy":
Steps:
1. Edit policy not attached to location
2. Edit policy attached to location
3. ...
The third Test is "Delete ACL policy":
Steps:
1. "Delete ACL Profile with Name: 'test_acl_valid' (Not attached to security policy)"
2. ...

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 8]
```diff
@@ -0,0 +1,655 @@
...
+                "Key": "tags",
+                "Value": "[\"one_tag\"]"
+              }
+            ]
+          },
+          "Expected": {
+            "Code": "200",
+            "Type": "json"
+          }
+        },
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:259]: Add validation step with method:GET and Response that checks tags:"[\"one_tag\"]" for this specific policy.

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 9]
```diff
@@ -0,0 +1,919 @@
...
+            ]
+          }
+        }
+      ]
+    },
+    {
+      "Name": "Delete Not Attached ACL Policy",
+      "Skip": false,
+      "Steps": [
+        {
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:404]: Add step to create unique acl profile to use in this test.

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 10]
```diff
@@ -0,0 +1,919 @@
...
+            ]
+          }
+        }
+      ]
+    },
+    {
+      "Name": "Delete Attached ACL Policy",
+      "Skip": false,
+      "Steps": [
+        {
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:431]: Add step to create unique acl profile to use in this test.

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 11]
```diff
@@ -0,0 +1,919 @@
...
+                "Key": "tags",
+                "Value": "[\"t\"]"
+              }
+            ]
+          },
+          "Expected": {
+            "Code": "400",
+            "Type": "json"
+          }
+        },
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:763]: Add test for tag with 2 chars.

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 12]
```diff
@@ -0,0 +1,1053 @@
...
+        }
+      ]
+    },
+    {
+      "Name": "Delete Not Attached ACL Policy",
+      "Skip": false,
+      "Steps": [
+        {
+          "Name": "Create ACL Profile with Trailing Space Characters in Name",
+          "Skip": false,
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:406]: Change name of the step: 
"Name": "Create ACL Profile with name 'not_attached_to_security_policy'"

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 13]
```diff
@@ -0,0 +1,1053 @@
...
+        }
+      ]
+    },
+    {
+      "Name": "Delete Attached ACL Policy",
+      "Skip": false,
+      "Steps": [
+        {
+          "Name": "Create ACL Profile with Trailing Space Characters in Name",
+          "Skip": false,
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:468]: Change name of the step:
"Name": "Create ACL Profile with name 'attached_to_security_policy'"

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 14]
```diff
@@ -0,0 +1,1053 @@
...
+            ]
+          },
+          "Expected": {
+            "Code": "400",
+            "Type": "json"
+          }
+        },
+        {
+          "Name": "Create ACL Profile with Trailing Space Characters in Name",
+          "Skip": false,
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:932]: Change name of the step:
"Name": "Create ACL Profile with Trailing Space Characters in Tag"

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 15]
```diff
@@ -0,0 +1,1053 @@
...
+          "Method": "GET",
+          "Expected": {
+            "Code": "200",
+            "Type": "json",
+            "Single": true,
+            "SingleResult": [
+              {
+                "Key": "name",
+                "Value": "\"acl_profile_tag_with_trailing_spaces\""
+              }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:986]: The assertion should be for tag, not for name

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 16]
```diff
@@ -0,0 +1,1053 @@
...
+            ]
+          }
+        },
+        {
+          "Name": "Create ACL Profile with Leading Space Characters in Name",
+          "Skip": false,
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:992]: Change name of the step:
"Name": "Create ACL Profile with Leading Space Characters in Tag"

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 17]
```diff
@@ -0,0 +1,1053 @@
...
+          "Method": "GET",
+          "Expected": {
+            "Code": "200",
+            "Type": "json",
+            "Single": true,
+            "SingleResult": [
+              {
+                "Key": "name",
+                "Value": "\"acl_profile_tag_with_leading_spaces\""
+              }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:1046]: The assertion should be for tag, not for name

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 18]
```diff
@@ -0,0 +1,1053 @@
...
+          "API": {
+            "Base": "acl-profile",
+            "Path": "__acldefault__/"
+          },
+          "Method": "DELETE",
+          "Expected": {
+            "Code": "400",
+            "Type": "json"
+          }
+        }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:331]: Expected should be:
"Expected": {
            "Code": "400",
            "Type": "content",
            "Single": true,
            "Text": "b\"schema mismatched: \\n'' is too short\\n\\nFailed validating 'minLength' in schema['properties']['name']:\\n    {'description': 'Name of entity shown in UI',\\n     'minLength': 1,\\n     'title': 'Name',\\n     'type': 'string'}\\n\\nOn instance['name']:\\n    ''\""
          }

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 19]
```diff
@@ -0,0 +1,1053 @@
...
+              {
+                "Key": "id",
+                "Value": "\"space_name\""
+              }
+            ]
+          },
+          "Expected": {
+            "Code": "400",
+            "Type": "json"
+          }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:198]: Expected should be:
"Expected": {
            "Code": "400",
            "Type": "content",
            "Single": true,
            "Text": "b\"schema mismatched: \\n'' is too short\\n\\nFailed validating 'minLength' in schema['properties']['name']:\\n    {'description': 'Name of entity shown in UI',\\n     'minLength': 1,\\n     'title': 'Name',\\n     'type': 'string'}\\n\\nOn instance['name']:\\n    ''\""
          }

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 20]
```diff
@@ -0,0 +1,1053 @@
...
+          "Expected": {
+            "Code": "200",
+            "Type": "json",
+            "Single": true,
+            "SingleResult": [
+              {
+                "Key": "tags",
+                "Value": "[\"one_tag\"]"
+              }
+            ]
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:395]: SingleResult should be:
"SingleResult": [
              {
                "Key": "tags",
                "Value": "['one_tag']"
              }
            ]

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 21]
```diff
@@ -0,0 +1,1053 @@
...
+          "Expected": {
+            "Code": "200",
+            "Type": "json",
+            "Single": true,
+            "SingleResult": [
+              {
+                "Key": "name",
+                "Value": "\"space_name_with_trailing_spaces\""
+              }
+            ]
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:309]: Should be:
"SingleResult": [
              {
                "Key": "name",
                "Value": "space_name_with_trailing_spaces"
              }
            ]

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 22]
```diff
@@ -0,0 +1,1053 @@
...
+            "Code": "200",
+            "Type": "json",
+            "Single": true,
+            "SingleResult": [
+              {
+                "Key": "name",
+                "Value": "\"space_name_with_spaces\""
+              }
+            ]
+          }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:254]: Should be: (no need for additional double quotes)
"SingleResult": [
              {
                "Key": "name",
                "Value": "space_name_with_spaces"
              }
            ]

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 23]
```diff
@@ -0,0 +1,1053 @@
...
+          "API": {
+            "Base": "acl-profile",
+            "Path": "attached_to_security_policy/"
+          },
+          "Method": "DELETE",
+          "Expected": {
+            "Code": "400",
+            "Type": "json"
+          }
+        }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:547]: Should be:
"Expected": {
            "Code": "400",
            "Type": "content",
            "Single": true,
            "Text": "b''"
          }

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 24]
```diff
@@ -0,0 +1,1053 @@
...
+              {
+                "Key": "tags",
+                "Value": "[\"a\"]"
+              }
+            ]
+          },
+          "Expected": {
+            "Code": "400",
+            "Type": "json"
+          }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:793]: "Expected": {
"Code": "400",
"Type": "content",
"Single": true,
"Text": "b''"
}

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 25]
```diff
@@ -0,0 +1,1053 @@
...
+              {
+                "Key": "tags",
+                "Value": "[\"aa\"]"
+              }
+            ]
+          },
+          "Expected": {
+            "Code": "400",
+            "Type": "json"
+          }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:825]: "Expected": {
"Code": "400",
"Type": "content",
"Single": true,
"Text": "b''"
}

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 26]
```diff
@@ -0,0 +1,1053 @@
...
+              {
+                "Key": "tags",
+                "Value": "[\"t\"]"
+              }
+            ]
+          },
+          "Expected": {
+            "Code": "400",
+            "Type": "json"
+          }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:896]: "Expected": {
"Code": "400",
"Type": "content",
"Single": true,
"Text": "b''"
}

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 27]
```diff
@@ -0,0 +1,1053 @@
...
+              {
+                "Key": "tags",
+                "Value": "[\"   \"]"
+              }
+            ]
+          },
+          "Expected": {
+            "Code": "400",
+            "Type": "json"
+          }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:928]: "Expected": {
"Code": "400",
"Type": "content",
"Single": true,
"Text": "b''"
}

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 28]
```diff
@@ -0,0 +1,1053 @@
...
+          "Method": "GET",
+          "Expected": {
+            "Code": "200",
+            "Type": "json",
+            "Single": true,
+            "SingleResult": [
+              {
+                "Key": "tags",
+                "Value": "[\"tag_with_trailing_spaces\"]"
+              }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:986]: Should be:
"SingleResult": [
              {
                "Key": "tags",
                "Value": "['tag_with_trailing_spaces']"
              }

File: example_suites/add_edit_duplicate_delete_acl_policy.json [hunk 29]
```diff
@@ -0,0 +1,1053 @@
...
+          "Method": "GET",
+          "Expected": {
+            "Code": "200",
+            "Type": "json",
+            "Single": true,
+            "SingleResult": [
+              {
+                "Key": "tags",
+                "Value": "[\"tag_with_leading_spaces\"]"
+              }
```
Comment (olgasobolevsky) [inline example_suites/add_edit_duplicate_delete_acl_policy.json:1046]: Should be:
"SingleResult": [
{
"Key": "tags",
"Value": "['tag_with_leading_spaces']"
}

**Pull request closed without merging**